### PR TITLE
PE: Create pe_opencaster for Opencaster streams

### LINKED ISSUE #2991: VZTV
**Channel Name:** VZTV
**Channel ID:** VZTV.pe
**Country:** Peru
**Language:** Spanish
**External Link (optional):** https://moyobamba.com/vztv/
**Notes (optional):** Not 24/7. Channel with "strict-origin-when-cross-origin" policy from opencaster. Iframe link: https://www.opencaster.com/abelhoyos3-100-100.

### LINKED ISSUE #4563: TV5 Soritor
**Channel Name:** TV5 Soritor
**Country:** Peru
**Language:** Spanish
**Website:** https://moyobamba.com/tv5soritor/
**Notes:** Opencaster streaming

### LINKED ISSUE #4567: Canal 21 Huancayo
**Channel Name:** Canal 21 Huancayo
**Channel ID:** Canal21Huancayo.pe
**Country:** Peru
**Language:** Spanish
**Website:** https://canal21huancayo.com/en-vivo/
**Notes:** Opencaster streaming

### LINKED ISSUE #4575: Milenia TV
**Channel Name:** Milenia TV
**Channel ID:** MileniaTVHD.pe
**Country:** Perú
**Language:** Spanish
**Website:** http://www.radiomileniaperu.com/mileniatv/
**Notes (optional):** Opencaster streaming

### LINKED ISSUE #4914: TV Surymar
**Channel Name:** TV Surymar
**Channel ID:** TVSurymar.pe
**Country:** PE
**Language:** Spanish
**Website:** http://www.televisionsurymar.com/ (updated)
**Stream URL (optional):** Unknow (see below)
**Notes (optional):** This channel has a app for stream based of Opencaster: https://play.google.com/store/apps/details?id=radio.tvsurymar for live tv. And social network: https://m.facebook.com/televisionsurymar/

### LINKED ISSUE #4990: TVO Piura
**Channel Name:** TVO Piura
**Channel ID:** TVOPiura.pe
**Country:** PE
**Language:** Spanish
**Website:** https://tvopiura.com/tvonline/
**Stream URL (optional):** https://www.opencaster.com/tvocanal31-100-100
**Notes (optional):** Opencaster embed.

### LINKED ISSUE #4991: Tele Vida Moyobamba
**Channel Name:** TeleVida Moyobamba
**Channel ID:** TeleVida.pe
**Country:** PE
**Language:** Spanish
**Website:** -- (Facebook is https://es-la.facebook.com/TeleVidaMoyobamba/)
**Stream URL (optional):** http://www.opencaster.com/televidatv
**Notes (optional):** Opencaster embed.

### PR DESCRIPTION
**Pull request by sguinetti**
This is a draft for adding new streams via Opencaster (http://www.opencaster.com/directory-0-1). Would be fix several channel requests.

Fixes #4567 Fixes #4914 Fixes #4991 Fixes #4990 Fixes #4563 Fixes #2991 and Fixes #4575

### CODE REVIEW

**Comment by LaneSh4d0w** [comment]
https://ed10ov.live.opencaster.com/hudxpzriGFDa/index.m3u8

**Comment by LaneSh4d0w** [comment]
Radio Estudio C Lima : http://158.69.252.195:8002/stream.m3u

File: channels/pe_opencaster.m3u [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#EXTM3U
+#EXTINF:-1 tvg-id="Cultura24tv.pe" tvg-country="PE" tvg-language="Spanish" tvg-logo="https://i.imgur.com/2mSwwpH.png" group-title="Culture",Cultura 24 (720p)
+https://vs8.live.opencaster.com/cultura24/smil:cultura24/playlist.m3u8
```
Comment (LaneSh4d0w) [inline channels/pe_opencaster.m3u:3]: Alternative : 
http://51.161.13.77:1935/cultura24/smil:cultura24/playlist.m3u8

**Comment by sguinetti** [comment]
> https://ed10ov.live.opencaster.com/hudxpzriGFDa/index.m3u8

Yes. The website is https://www.radiomisionerafuentedevida.net/

**Comment by sguinetti** [comment]
I see this [video from Youtube](https://www.youtube.com/watch?v=). In 12:57, this person talks about url stream like https://live.opencaster.com/opencaster/code/playlist.m3u8.

**Comment by LaneSh4d0w** [comment]
Link isn't valid.

**Comment by sguinetti** [comment]
Apparently, only administrators can share such links.

**Comment by LaneSh4d0w** [comment]
We can't make those links work for now.

**Comment by GiovanniSebastian** [comment]
Works with TVES (Venezuela)?

**Comment by LaneSh4d0w** [comment]
For those who want to use it locally : https://github.com/LaneSh4d0w/streamlink/blob/master/src/streamlink/plugins/opencaster.py

**Comment by freearhey** [comment]
The last commit only resolves the conflict caused by updating the repository structure.

**Review by its0x08 — DISMISSED** [review]

**Comment by freearhey** [comment]
@sguinetti Do you plan to make any further changes to this request?

**Comment by LaneSh4d0w** [comment]
> @sguinetti Do you plan to make any further changes to this request?

To be honest we can't do much. Opencaster links are origin-locked and we can't change that header, since it's protected on every browser. Maybe we could add this on a "Kodi-specific" playlist, if we plan to create this in the future.

**Comment by sguinetti** [comment]
The best solution to operate this streams is to use your script and implement it in streamlink to use as a link like this:

http://free.fullspeed.tv/iptv-query?streaming-ip=http://www.opencaster.com/nicolezevallos98

**Comment by LaneSh4d0w** [comment]
This can only be used locally.

**Comment by GiovanniSebastian** [comment]
This Work? http://vs8.live.opencaster.com/20100152275/jcpstream/playlist.m3u8

**Comment by LaneSh4d0w** [comment]
> This Work? http://vs8.live.opencaster.com/20100152275/jcpstream/playlist.m3u8

Yes

**Comment by LaneSh4d0w** [comment]
Added Television Venezolana Social (TVes), officially using Opencaster (and OK.ru) for their streams.

https://www.tves.gob.ve/en-vivo/

**Review by its0x08 — APPROVED** [review]

**Comment by LaneSh4d0w** [comment]
Closed this PR since it's obvious we won't make them work for now. However, if Opencaster changes their strategy, we might reopen this PR. Not removing the branch since it contains lots of channel additions and moves.

**Pull request closed without merging**